pkg/gadgets/internal/socketenricher/bpf/sockets-map.bpf.c

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: (GPL-2.0 WITH Linux-syscall-note) OR Apache-2.0
+/* Copyright (c) 2021 The Inspektor Gadget authors */
+
+#include <vmlinux/vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+
+#include <bpf/bpf_helpers.h>
+
+#define SOCKETS_MAP_IMPLEMENTATION
+#include "sockets-map.h"
+
+#define MAX_ENTRIES	10240
+
+// The map 'start' keeps context between a kprobe and a kretprobe
+// Keys: pid_tgid
+// Values: the argument of the kprobe function:
+// - When used in bind: struct socket *
+// - When used in tcp_connect: struct sock *
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, __u64);
+	__type(value, void *);
+} start SEC(".maps");
+
+static __always_inline void
+update_sockets_map(struct sock *sock)
+{
+	struct sockets_key socket_key = {0,};
+	struct inet_sock *inet_sock = (struct inet_sock *)sock;
+	BPF_CORE_READ_INTO(&socket_key.netns, sock, __sk_common.skc_net.net, ns.inum);
+	socket_key.proto = BPF_CORE_READ(sock, sk_protocol);
+	socket_key.port = bpf_ntohs(BPF_CORE_READ(inet_sock, inet_sport));
```

Comment:
I'm (for the nth time in the professional career) confused by endianess. In which endianess are ports saved on the sockets map? (host or network?)

If I use bpftool, I suppose it's saved in host as I can see `"port": 8000` after using `dig -b 172.17.0.3#8000 google.com`, but then, how does it work if there is not any endianess conversion in `gadget_socket_lookup()`?

Reply:
In the kernel, `struct inet_sock` and `struct sock_common` have the following fields:
```
__be16 inet_sport;
__be16 skc_dport;
```
`__be16` => network endiannnes. Then we convert to host with bpf_ntohs, so the map stores values in host.

It is confusing because the kernel uses both host and network endianness in different fields...

Reply:
I think I should add a comment near the map definition and where it is assigned.

Reply:
> The eBPF instructions load_half and load_word used in gadget_socket_lookup() automatically convert from network to host endianness 

:open_mouth: I didn't know it, now everything makes sense. 

I wondering if this makes sense to save them in networking order to avoid the endianess conversion for each packet, but I don't know how costly it is. Anyway, a comment could help a lot :D.